plugins/modules/fabric_virtual_networks_workflow_manager.py

```diff
@@ -0,0 +1,2950 @@
+        }
+
+        if fabric_locations:
+            fabric_ids = self.get_fabric_ids(fabric_locations)
+
+            if fabric_ids:
+                vn_payload["fabricIds"] = fabric_ids
+
+        site_name = vn_detail.get("anchored_site_name")
+        if site_name:
+            site_id = self.get_site_id(site_name)
+            if not site_id:
+                msg = "Given Anchor site '{0}' not  present in Cisco Catalyst Center.".format(site_name)
+                self.log(msg, "ERROR")
+                return vn_payload
+            try:
+                anchor_fabric_id = self.get_fabric_site_id(site_name, site_id)
+            except Exception as e:
+                anchor_fabric_id = self.get_fabric_zone_id(site_name, site_id)
+
+            if anchor_fabric_id:
+                vn_payload["anchoredSiteId"] = anchor_fabric_id
+
+        return vn_payload
+
+    def get_vn_details_from_ccc(self, vn_name):
+        """
+        Retrieves details of a specified Layer3 Virtual Network from the Cisco Catalyst Center.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            vn_name (str): The name of the Layer3 Virtual Network whose details are to be fetched.
+        Returns:
+            dict or None: A dictionary containing the details of the specified virtual network if found;
+                        otherwise, returns None.
+        Description:
+            This function queries the Cisco Catalyst Center API to obtain information about a Layer3
+            Virtual Network identified by the provided name.
+            If the response does not contain any data, it logs an informational message indicating that the
+            specified virtual network is not present.
+            The function returns the details of the virtual network as a dictionary, or None if the network
+            does not exist or if an error occurs during the retrieval process.
+        """
+
+        try:
+            response = self.dnac._exec(
+                family="sda",
+                function='get_layer3_virtual_networks',
+                op_modifies=True,
+                params={
+                    "virtual_network_name": vn_name,
+                },
+            )
+            response = response.get("response")
+            self.log("Received API response from 'get_layer3_virtual_networks' for the vn '{0}': {1}".format(vn_name, str(response)), "DEBUG")
+
+            if not response:
+                self.log("Given layer3 virtual network '{0}' is not present in Cisco Catalyst Center.".format(vn_name), "INFO")
+                return None
+
+        except Exception as e:
+            self.msg = (
+                "Error while getting the details for layer3 virtual network '{0}' present in "
+                "Cisco Catalyst Center: {1}"
+            ).format(vn_name, str(e))
+            self.set_operation_result("failed", False, self.msg, "ERROR").check_return_status()
+
```

Comment:
self.log("Details retrieved successfully for Layer3 virtual network '{0}': {1}".format(vn_name, response[0]), "INFO")